### ISSUE TITLE
Declaration of ilMyTableGUI::addActionButton() should be compatible with ilTableGUI

### ISSUE THREAD

**Issue by fsesterh** [hrzbonn/Todolists#5]
Hello,
ILIAS 5.3.0 stable was released yesterday, so I had a look at the plugin and tried if it works with the new version. To do so I increased the version number in plugin.php to "5.3.999".

Unfortunately I ran into this error, when accessing the mileston tab:
Declaration of ilMyTableGUI::addActionButton() should be compatible with ilTableGUI

#10 Whoops\Exception\ErrorException in /opt/rh/httpd24/root/var/www/html/core/Customizing/global/plugins/Services/Repository/RepositoryObject/Todolists/classes/class.ilMyTableGUI.php:13
#9 ilErrorHandling:handlePreWhoops in /opt/rh/httpd24/root/var/www/html/core/Customizing/global/plugins/Services/Repository/RepositoryObject/Todolists/classes/class.ilMyTableGUI.php:13
#8 include_once in /opt/rh/httpd24/root/var/www/html/core/Customizing/global/plugins/Services/Repository/RepositoryObject/Todolists/classes/class.ilMilestoneListGUI.php:3
#7 include_once in /opt/rh/httpd24/root/var/www/html/core/Customizing/global/plugins/Services/Repository/RepositoryObject/Todolists/classes/class.ilObjTodolistsGUI.php:1908
#6 ilObjTodolistsGUI:milestone in /opt/rh/httpd24/root/var/www/html/core/Customizing/global/plugins/Services/Repository/RepositoryObject/Todolists/classes/class.ilObjTodolistsGUI.php:107
#5 ilObjTodolistsGUI:performCommand in /opt/rh/httpd24/root/var/www/html/core/Services/Repository/classes/class.ilObjectPluginGUI.php:174
#4 ilObjectPluginGUI:executeCommand in /opt/rh/httpd24/root/var/www/html/core/Services/UICore/classes/class.ilCtrl.php:197
#3 ilCtrl:forwardCommand in /opt/rh/httpd24/root/var/www/html/core/Services/Repository/classes/class.ilObjPluginDispatchGUI.php:46
#2 ilObjPluginDispatchGUI:executeCommand in /opt/rh/httpd24/root/var/www/html/core/Services/UICore/classes/class.ilCtrl.php:197
#1 ilCtrl:forwardCommand in /opt/rh/httpd24/root/var/www/html/core/Services/UICore/classes/class.ilCtrl.php:159
#0 ilCtrl:callBaseClass in /opt/rh/httpd24/root/var/www/html/core/ilias.php:21

Steps to reproduce:
================
- create a new todo list
- access the milestone tab

Since it might be important: The development machine I am testing on is running PHP7.1.

We're aiming to update our productive environment mid March, so hopefully this can be fixed till then.

Thanks in advance!
Regards
Fabian

**Comment by hrzbonn** [comment]
Hello,
we try to fix this problem until mid march. But we are still working on an internal project which also should finished until march. It is possible that we can not finished both. It depends on how much time the fix needs. We hope you can understand this.

**Comment by hrzbonn** [comment]
Hello again,
i try to reproduce the error but it does not occur. Did you have done any other step? Or have you another hint for me to reproduce the error?

**Comment by fsesterh** [comment]
Not really. What PHP version are you using? I run 7.1.8 (supported since ILIAS 5.3).

**Comment by hrzbonn** [comment]
Sorry for the delay in my reply. We are still working to reproduce the error. At this Time we are not using PHP 7.1.8 but we are working to set up this PHP-Version.

**Comment by fsesterh** [comment]
In the meantime I downgrades to PHP7.0. Same error there :(

[qv2fe_3145.log](https://github.com/hrzbonn/Todolists/files/1767089/qv2fe_3145.log)

**Comment by hrzbonn** [comment]
I'm sorry that we need so much time, but the error is now fixed. You just need to update the new version of the plugin. Thank you for reporting the bug and using our plugin.
